src/rgw/driver/rados/rgw_notify.cc

```diff
@@ -711,7 +711,11 @@ static inline void populate_event(reservation_t& res,
     // either no metadata exist or no metadata filter was used
     metadata_from_attributes(res, obj);
   }
-  event.x_meta_map = res.x_meta_map;
+  std::copy_if(res.x_meta_map.cbegin(), res.x_meta_map.cend(),
+               std::inserter(event.x_meta_map, event.x_meta_map.end()),
+               [](const auto& m) {
+                 return (m.first.rfind("x-amz-meta-", 0) == 0);
```

Comment:
the condition for being a metadata attribute is in line 646:
```
boost::algorithm::starts_with(attr.first, RGW_ATTR_META_PREFIX)
```

also, not clear to me how these attributes get into the reservation object in the first place. since the above condition should make sure only metadata attributes are inserted to the map

Reply:
if the map isn't general, then yes, it would be preferable to not add them ;)

Reply:
made the changes to stop populating non-metadata attributes to the reservation object